sample-app/src/components/Navigation.tsx

```diff
@@ -2,18 +2,43 @@ import classNames from 'classnames';
 import { useCallback, useEffect, useLayoutEffect, useRef, useState } from 'react';
 import { NavLink, useLocation } from 'react-router-dom';
 import { ReactComponent as KebabIcon } from '../icons/kebab.svg';
-import '../sass/Navigation.scss';
+import { useComposedCssClasses, CompositionMethod } from '../hooks/useComposedCssClasses';
+
+interface NavigationCssClasses {
+  nav?: string,
+  linksWrapper?: string,
+  menuWrapper?: string,
+  navLink?: string,
+  activeNavLink?: string,
+  menuButton?: string,
+  menuButtonContainer?: string,
+  menuButton___menuOpen?: string,
+  menuContainer?: string
+}
+
+const builtInCssClasses: NavigationCssClasses = {
+  nav: 'border-b border-gray-200 text-gray-600 flex space-x-6 font-medium pt-4',
+  navLink: 'whitespace-nowrap py-4 px-1 font-medium text-md border-b-2 border-opacity-0 hover:border-gray-300',
+  activeNavLink: 'text-blue-600 border-blue-600 border-b-2 border-opacity-100 hover:border-blue-600',
+  menuButtonContainer: 'relative flex flex-grow justify-end mr-4',
+  menuButton: 'flex flex-row items-center font-medium text-md px-1 border-b-2 border-opacity-0 hover:border-gray-300',
+  menuButton___menuOpen: 'bg-gray-200',
```

Comment:
is the triple underscore intentional? I would have expected just 2 underscores

Reply:
In a traditional BEM setup, there would be two dashes, however two dashes isn't valid as an object key without putting it in quotes. We could do that, but then we would also need to change how we access the classNames from `cssClasses.menuButton___menuOpen` to `cssClasses['menuButton--menuOpen']`. I would prefer to avoid having that edge case in the code, but it's an option. A solution that I've seen before is to update BEM from `Block__element--modifier` to `Block_element___modifier`. I was using this convention by putting three underscores before the modifier. However two underscores would also work

Reply:
ok got it! was just wondering if it was a typo, 3 is fine